common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -975,24 +975,156 @@ def _get_section_score(self):
+
+    def _set_staff_lock_on_subsection(self, locked):
+        """
+        Sets staff lock for a subsection, which should hide the
+        subsection score from students on the progress page.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            subsection = self.course_outline.section_at(0).subsection_at(0)
+            subsection.set_staff_lock(locked)
+            self.assertEqual(subsection.has_staff_lock_warning, locked)
+
+    def _change_weight_for_problem(self):
+        """
+        Changes the weight of the problem, which should not affect
+        persisted grades.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            self.course_outline.section_at(0).subsection_at(0).expand_subsection()
+            unit = self.course_outline.section_at(0).subsection_at(0).unit(self.UNIT_NAME).go_to()
+            container = unit.xblocks[0].go_to_container()
+            component = container.xblocks[0].children[0]
+
+            component.edit()
+            component_editor = ComponentEditorView(self.browser, component.locator)
+            component_editor.set_field_value_and_save('Problem Weight', 5)
+
+    def _rescore_for_all(self):
+        """
+        Triggers a rescore for all students. Currently unimplemented.
+        """
+        pass
+
+    def _edit_problem_content(self):
+        """
+        Replaces the content of a problem with other html.
+        Should not affect persisted grades.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            self.course_outline.section_at(0).subsection_at(0).expand_subsection()
+            unit = self.course_outline.section_at(0).subsection_at(0).unit(self.UNIT_NAME).go_to()
+            component = unit.xblocks[1]
+            edit_view = component.edit()
+
+            modified_content = "<p>modified content</p>"
+            # Set content in the CodeMirror editor.
+            type_in_codemirror(self, 0, modified_content)
+
+            edit_view.q(css='.action-save').click()
+
+    @ddt.data(
+        _edit_problem_content,
+        _change_subsection_structure,
+        _change_weight_for_problem,
+        _rescore_for_all
+    )
+    def test_content_changes_do_not_change_score(self, edit):
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(0, 1)])
+            self.assertEqual(self._get_section_score(), (0, 1))
+            self.courseware_page.visit()
+            self._answer_problem_correctly()
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+        edit(self)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_scores(), [(1, 1)])
+            self.assertEqual(self._get_section_score(), (1, 1))
+
+    def test_staff_lock(self):
```

Comment:
Nit: to be consistent with the above, consider renaming `test_visibility_change_does_affect_score`?